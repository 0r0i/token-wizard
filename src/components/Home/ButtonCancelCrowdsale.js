import React, { Component } from 'react'
import logdown from 'logdown'
import cancelDeploy from '../../utils/cancelDeploy'

const logger = logdown('TW:Home:CancelCrowdsale')

export class ButtonCancelCrowdsale extends Component {
  cancel = () => {
    logger.log('Cancel crowdsale')
    cancelDeploy()
  }

  render() {
    return (
      <button onClick={() => this.cancel()} className="hm-ButtonCancelCrowdsale">
        Cancel crowdsale
      </button>
    )
  }
}
