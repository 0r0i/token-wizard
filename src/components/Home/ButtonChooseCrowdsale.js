import React, { Component } from 'react'
import { navigateTo } from '../../utils/utils'

export class ButtonChooseCrowdsale extends Component {
  goToCrowdsales = () => {
    localStorage.reload = true
    navigateTo({
      history: this.props.history,
      location: 'crowdsales',
      fromLocation: 'home'
    })
  }

  render() {
    return (
      <div onClick={() => this.goToCrowdsales()} className="hm-ButtonChooseCrowdsale">
        Choose Crowdsale
      </div>
    )
  }
}
