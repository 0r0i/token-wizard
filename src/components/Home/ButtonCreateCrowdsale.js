import React, { Component } from 'react'
import { navigateTo } from '../../utils/utils'

export class ButtonCreateCrowdsale extends Component {
  goToStepOne = () => {
    localStorage.reload = true
    navigateTo({
      history: this.props.history,
      location: 'stepOne',
      fromLocation: 'home'
    })
  }

  render() {
    return (
      <button onClick={() => this.goToStepOne()} className="hm-ButtonCreateCrowdsale">
        New crowdsale
      </button>
    )
  }
}
