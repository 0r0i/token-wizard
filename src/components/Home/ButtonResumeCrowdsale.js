import React, { Component } from 'react'

export class ButtonResumeCrowdsale extends Component {
  resume = () => {
    localStorage.reload = true
    window.location = '/4'
  }

  render() {
    return (
      <button onClick={() => this.resume()} className="hm-ButtonResumeCrowdsale">
        Resume crowdsale
      </button>
    )
  }
}
