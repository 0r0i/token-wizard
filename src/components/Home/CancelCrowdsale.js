import React, { Component } from 'react'
import logdown from 'logdown'
import cancelDeploy from '../../utils/cancelDeploy'

const logger = logdown('TW:Home:CancelCrowdsale')

export class CancelCrowdsale extends Component {
  cancel = () => {
    logger.log('Cancel crowdsale')
    cancelDeploy()
  }

  render() {
    return (
      <button onClick={() => this.cancel()} className="hm-Home_BtnChoose">
        Cancel crowdsale
      </button>
    )
  }
}
