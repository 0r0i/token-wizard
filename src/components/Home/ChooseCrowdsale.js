import React, { Component } from 'react'
import { navigateTo } from '../../utils/utils'

export class ChooseCrowdsale extends Component {
  goToCrowdsales = () => {
    navigateTo(this.props, 'crowdsales')
  }

  render() {
    return (
      <div onClick={() => this.goToCrowdsales()} className="hm-Home_BtnChoose">
        Choose Crowdsale
      </div>
    )
  }
}
