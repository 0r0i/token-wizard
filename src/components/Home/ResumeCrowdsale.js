import React, { Component } from 'react'
import logdown from 'logdown'
import { navigateTo } from '../../utils/utils'

const logger = logdown('TW:Home:ResumeCrowdsale')

export class ResumeCrowdsale extends Component {
  resume = () => {
    logger.log('Resume crowdsale')

    navigateTo({
      history: this.props.history,
      location: 'stepFour'
    })
  }

  render() {
    return (
      <button onClick={() => this.resume()} className="hm-Home_BtnNew">
        Resume crowdsale
      </button>
    )
  }
}
