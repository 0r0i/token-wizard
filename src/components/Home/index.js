import React, { Component } from 'react'
import { LogoPrimary } from '../LogoPrimary/index'
import { ChooseCrowdsale } from './ChooseCrowdsale'
import { CreateCrowdsale } from './CreateCrowdsale'
import { ResumeCrowdsale } from './ResumeCrowdsale'
import { CancelCrowdsale } from './CancelCrowdsale'
import { inject, observer } from 'mobx-react/index'

@inject('deploymentStore')
@observer
export class Home extends Component {
  render() {
    const { history, deploymentStore } = this.props

    return (
      <div>
        <section className={`hm-Home`}>
          <div className="hm-Home_Scroll">
            <div className="hm-Home_Top">
              <LogoPrimary />
            </div>
            <div className="hm-Home_MainInfoContainer">
              <div className="hm-Home_MainInfo">
                <div className="hm-Home_MainInfoTextContainer">
                  <h1 className="hm-Home_MainInfoTitle">Welcome to Token Wizard</h1>
                  <p className="hm-Home_MainInfoDescription">
                    Token Wizard is a client side tool to create ERC20 token and crowdsale in five steps. It helps you
                    to publish contracts on the Ethereum network, create a crowdsale page with stats. For participants,
                    the wizard creates a page to contribute into the campaign.
                  </p>
                  <p className="hm-Home_MainInfoPoweredBy">
                    Token Wizard is powered by <a href="https://github.com/auth-os/beta">Auth-os</a>.
                  </p>
                </div>
                {deploymentStore.deployInProgress ? (
                  <div className="hm-Home_MainInfoButtonContainer">
                    <ResumeCrowdsale history={history} />
                    <CancelCrowdsale history={history} />
                  </div>
                ) : (
                  <div className="hm-Home_MainInfoButtonContainer">
                    <CreateCrowdsale history={history} />
                    <ChooseCrowdsale history={history} />
                  </div>
                )}
              </div>
            </div>
          </div>
        </section>
      </div>
    )
  }
}
