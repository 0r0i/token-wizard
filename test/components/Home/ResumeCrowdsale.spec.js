import React from 'react'
import Adapter from 'enzyme-adapter-react-15'
import { configure, mount, shallow, render } from 'enzyme'
import { MemoryRouter } from 'react-router'
import { ResumeCrowdsale } from '../../../src/components/Home/ResumeCrowdsale'

configure({ adapter: new Adapter() })

describe('ResumeCrowdsale', () => {
  const history = {
    push: jest.fn()
  }

  it('should render screen with shallow without throwing an error', () => {
    const wrapper = shallow(<ResumeCrowdsale />)

    expect(wrapper).toMatchSnapshot()
  })

  it('should render screen with mount without throwing an error', () => {
    // Given
    const wrapper = mount(
      <MemoryRouter initialEntries={['/']}>
        <ResumeCrowdsale />
      </MemoryRouter>
    )

    // When
    const tree = wrapper.find('.hm-Home_BtnNew')

    // Then
    expect(tree).toHaveLength(1)
  })

  it('should render screen with render without throwing an error', () => {
    // Given
    const wrapper = render(
      <MemoryRouter initialEntries={['/']}>
        <ResumeCrowdsale />
      </MemoryRouter>
    )
    // When
    const tree = wrapper.html()

    // Then
    expect(tree).toMatchSnapshot()
  })

  it(`should simulate click`, () => {
    // Given
    const wrapper = shallow(<ResumeCrowdsale history={history} />)
    const instance = wrapper.instance()
    const resumeHandler = jest.spyOn(instance, 'resume')
    instance.forceUpdate()
    // When
    const buttonResumeCrowdsale = wrapper.find('.hm-Home_BtnNew')
    buttonResumeCrowdsale.simulate('click')

    // Then
    expect(buttonResumeCrowdsale.length).toBe(1)
    expect(resumeHandler).toHaveBeenCalledTimes(1)
  })
})
