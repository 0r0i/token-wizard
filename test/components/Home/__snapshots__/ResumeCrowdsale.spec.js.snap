// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ResumeCrowdsale should render screen with render without throwing an error 1`] = `"Resume crowdsale"`;

exports[`ResumeCrowdsale should render screen with shallow without throwing an error 1`] = `
<button
  className="hm-Home_BtnNew"
  onClick={[Function]}
>
  Resume crowdsale
</button>
`;
